src/harness/unittests/compileOnSave.ts

```diff
@@ -588,5 +588,37 @@ namespace ts.projectSystem {
             assert.isTrue(outFileContent.indexOf(file2.content) === -1);
             assert.isTrue(outFileContent.indexOf(file3.content) === -1);
         });
+
+        it("should emit js files in inferred projects when they could be input files", () => {
+            const f1 = {
+                path: "/a/b/f1.ts",
+                content: "var test = \"\";"
+            };
+            const f1js = {
+                path: "/a/b/f1.js",

```

Comment:
I would be tempted to call file "not_f1.js", just to emphasize that the names don't have to match.  If someone tightens the check and the test starts failing, they'll know that it's the test that's out of date.